Temporal control documentation starting (intro and raster) (#6379)
qgis/QGIS-Documentation: docs/user_manual/temporal/temporal.rst

.. _`temporal`:

****************
Temporal Support
****************

.. only:: html

   .. contents::
      :local:

> 

Reviewer: Can we consider adding a short description of what temporal data or temporal control means? And how this information can be stored? We usually do this when we introduce new "concepts" (see mesh, vector, raster, vector tile (soon)).

QGIS can handle temporal control on loaded layers depending on the information
of the data. 

Different providers are supported:

* raster layers
* vector layers
* mesh layers
* WMST

Activating the temporal control support means setting up the temporal options on
the layer(s) and then enabling the temporal view QGIS-wide.

The temporal support is also available in the print layout: you can export time
based maps.

> Furthermore, many variables are added to the Field Calculator so that you can

Reviewer: the field calculator is quite restrictive. Better refer to expressions....

use them to change the appearance of the layer(s) depending on the current map
frame.

General concepts
================
Temporal control settings are available at different levels:

> * in the `Temporal` tab within the project properties :menuselection:`Project --> Properties`

Reviewer: ```suggestion
* in the :guilabel:`Temporal` tab within the :menuselection:`Project --> Properties` dialog
```
to avoid repetition

> * in the properties on the single layer(s)

Reviewer: And what is, if any, the relation between the project level temporal settings and the layer level's?
I think this is where we would explain the QGIS "concepts" as mentioned in the section title.

Author: This will be cleared when using vector layers

The whole temporal control is activated with the |temporal| icon located in the

> main toolbar. Once clicked the temporal panel will appear:

Reviewer: ```suggestion
:guilabel:`Map Navigation` toolbar. Once clicked the temporal panel will appear:
```

.. _temporal_controller:

.. figure:: img/temporal_controller_panel.png
   :align: center

   Temporal Controller Panel

> The temporal panel has 3 main icons:

Author: :)

* |indicatorTemporalOff| :guilabel:`Turn off temporal navigation`: all the temporal 

>   settings are diabled
> * |indicatorTemporalFixedRange| :guilabel:`Fixed range tamporal navigation`: a 
>   fixed time range is set and oly layers that fit this range will be diplayed in 

Reviewer: ```suggestion
  settings are disabled
* |indicatorTemporalFixedRange| :guilabel:`Fixed range temporal navigation`: a 
  fixed time range is set and only layers that fit this range will be displayed in
```

  the map
* |indicatorTemporalAnimated| :guilabel:`Animated tamporal navigation`: the map

>   will be animated depending on the layer settings and temporal customization

Author: yes right, not sure it is the right moment to describe it.

In order to link the visible layers with the time range you have first to tweak
the single layer temporal settings.

.. _raster_temporal:

Raster Layers
=============
Normally raster layers don't have any temporal metadata set. Therefore you have
to choose layer by layer the temporal settings like the start date and
eventually the end date:

> .. figure:: img/raster_layer_temporal_settings_1.png

Reviewer: Any chance you could remove the background (hide the layer before shooting the dialog)? It's quite ugly this way.

   :align: center

   Raster Layer Temporal Settings

In this example the raster layer has a ``Start date`` set to ``2013-06-30 00:00:00``
and no ``End Date``. It means that the raster will be convered by other rasters
that have a more recent start date set.

Once you have enabled the temporal control on the layer the

> |indicatorTemporalFixedRange| icon will be dislayed next to the layer name in

Reviewer: ```suggestion
|indicatorTemporalFixedRange| icon will be displayed next to the layer name in
```

the legend to remind you that the layer is temporal controlled:

.. figure:: img/temporal_icon_legend.png
   :align: center

   Temporal icon in the legend

> You can click on the icon to quickly go to the temporal layer settings.

Reviewer: ```suggestion
You can click on the icon to quickly go to the temporal settings of the layer.
```
instead?

> If you have many different raster layers loaded in the legend be sure to tweak

Reviewer: ```suggestion
If you have many different raster layers loaded in the project be sure to tweak
```

the temporal settings for each layer. In the following example the name of the

> layers describes the date.

Reviewer: could we provide some more info on the date?
```suggestion
layers describes the date at which they are set to display.
```
or something like that...

If you open the main temporal panel with the |temporal| button you can animate
the map with by clicking the |indicatorTemporalAnimated| :guilabel:`Animated
tamporal navigation` button.

Before to play the map you have to set the temporal options shown in the bottom

> part of the ``Animated temporal navigation`` panel:

Reviewer: ```suggestion
part of the :guilabel:`Animated temporal navigation` panel:
```

> * ``Range``: the starting and ending datetime range
> * ``Step``: how often the map should update

Reviewer: ```suggestion
* :guilabel:`Range`: the starting and ending datetime range
* :guilabel:`Step`: how often the map should update
```

.. figure:: img/animated_raster_options.png
   :align: center

   Animated Temporal Navigation Options

Once you have set the temporal options you can animate the map with the |play|
button. Use all the other buttons to pause, stop, rewind and forward the map:

.. figure:: img/play_bar.png
   :align: center

   Animated Temporal Navigation Buttons

If the |unchecked| :guilabel:`Loop` checkbox in the upper right part of the bar

> if checked the animation will be endlessly repeated.

Reviewer: ```suggestion
is checked the animation will be endlessly repeated.
```

.. _`temporal`:

****************
Temporal Support
****************

.. only:: html

   .. contents::
      :local:

QGIS can handle temporal control on loaded layers depending on the information
of the data. 

Different providers are supported:

* vector layers
* raster layers
* mesh

> * WMST

Author: tihs is actually wms-t (wmts are wms tile based wms-t are wms with time data)
